app/modules/kit-library/data/repository.php

```diff
@@ -179,14 +182,26 @@ public function remove_from_favorites( $id ) {
 	 */
 	private function get_kits_data( $force_api_request = false ) {
 		$data = get_transient( static::KITS_CACHE_KEY );
+		$cached_kits_editor_layout_type = get_transient( static::CACHED_KITS_LAYOUT_TYPE );
 
-		if ( ! $data || $force_api_request ) {
-			$data = $this->api->get_all();
+		$experiments_manager = Plugin::$instance->experiments;
+		$kits_editor_layout_type = $experiments_manager->is_feature_active( 'container' ) ? 'container_flexbox' : '';
+
+		if ( ! $data || $force_api_request || $cached_kits_editor_layout_type !== $kits_editor_layout_type ) {
+			$args = [
+				'body' => [
+					'editor_layout_type' => $kits_editor_layout_type,
+					'minimum_version' => ELEMENTOR_VERSION,
+				],
+			];
+
+			$data = $this->api->get_all( $args );
 
 			if ( is_wp_error( $data ) ) {
 				throw new WP_Error_Exception( $data );
 			}
 
+			set_transient( static::CACHED_KITS_LAYOUT_TYPE, $kits_editor_layout_type, DAY_IN_SECONDS );
```

Comment:
Not as a change request, but more as a thought - maybe worth saving both the data and the layout type in one transient, to prevent double reading/writing to the DB?

Reply:
Changed approach for the editor_layout_type.